include/ginkgo/core/matrix/dense.hpp

```diff
@@ -232,6 +232,16 @@ class Dense
     std::unique_ptr<LinOp> row_permute(
         const Array<int64> *permutation_indices) const override;
 
+    std::unique_ptr<Dense> row_gather(const Array<int32> *gather_indices) const;
```

Comment:
Since these are new and not an interface/override, documentation would be good. Also, I guess we expect to have this only with dense? Otherwise, how about creating an interface for this.

Reply:
I only added the minimal interface I need to make distributed SpMV work, so far I couldn't imagine where we would use this for other matrix formats or columns instead of rows.

Reply:
Maybe block CSR could find some use in something related at some point for a distributed block ParILU of some sort? But I guess it wouldn't be row gather but block gather? I guess indeed there is little application to the same interface elsewhere.